hooks/shotgun_filters.py

```diff
@@ -0,0 +1,222 @@
+
+        link_filters = []
+
+        if sg_location.entity_type in ["HumanUser"]:
+            # the logic for users is different
+            # here we want give an overview of their work
+            # for the current project
+
+            # When the current project is None, the user is in site context and we want to get
+            # the requested fields for all user's tasks, notes, versions and publishes.
+            if context_project:
+                link_filters.append(["project", "is", context_project])
+
+            if entity_type == "Task":
+                # show tasks i am assigned to
+                link_filters.append(["task_assignees", "in", [sg_location.entity_dict]])
+                link_filters.append(["sg_status_list", "is_not", "fin"])
+                if context_project:
+                    # we are in a non-site context. only tasks from this project
+                    link_filters.append(["project", "is", context_project])
+
+            elif (
+                entity_type == "Note"
+                and sg_location.entity_type == context_user.get("type")
+                and sg_location.entity_id == context_user.get("id")
+            ):
+                # not just any user, but this is ME!
+                # show notes that are TO me, CC me or on tasks which I have been
+                # assigned. Use advanced filters for this one so we can use OR
+                #
+                # we basically want to show notes that are FOR me.
+
+                link_filters.append(
+                    {
+                        "filter_operator": "or",
+                        "filters": [
+                            ["created_by", "is", sg_location.entity_dict],
+                            [
+                                "addressings_cc.Group.users",
+                                "in",
+                                sg_location.entity_dict,
+                            ],
+                            [
+                                "addressings_to.Group.users",
+                                "in",
+                                sg_location.entity_dict,
+                            ],
+                            ["replies.Reply.user", "is", sg_location.entity_dict],
+                            ["addressings_cc", "in", sg_location.entity_dict],
+                            ["addressings_to", "in", sg_location.entity_dict],
+                            [
+                                "tasks.Task.task_assignees",
+                                "in",
+                                sg_location.entity_dict,
+                            ],
+                        ],
+                    }
+                )
+
+                if context_project:
+                    # we are in a non-site context. only tasks from this project
+                    link_filters.append(["project", "is", context_project])
+
+            elif entity_type == "Note":
+                # another user that isn't me.
```

Comment:
What's this mean? It's confusing when taking 102-104 into consideration.

Reply:
Not sure to be honest. I'll update this comment